components/homme/src/share/cxx/PpmRemap.hpp

```diff
@@ -296,38 +308,33 @@ template <typename boundaries> struct PpmVertRemap : public VertRemapAlg {
     // Then take the difference between accumulation at successive interfaces
     // gives the mass inside each cell. Since Qdp is supposed to hold the full
     // mass this needs no normalization.
+    Real mass1 = 0;
+    Real mass2;
+    ExecViewUnmanaged<Real[NUM_PHYSICAL_LEV]> rvar(reinterpret_cast<Real*>(remap_var.data()));
     Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team, NUM_PHYSICAL_LEV),
                          [&](const int k) {
       const int kk_cur_lev = k_id(k);
-      assert(kk_cur_lev + 1 >= k);
       assert(kk_cur_lev < parabola_coeffs.extent_int(1));
 
       const Real x2_cur_lev = integral_bounds(k);
       // Repurpose the mass buffer to store the new mass.
       // WARNING: This may not be thread safe in future architectures which
       //          use this level of parallelism!!!
-      mass(k) = compute_mass(
+      mass2 = compute_mass(
```

Comment:
I don't think this can work on GPU, in general. mass(k) is being updated in parallel (vector range). So one can't then do `mass1 = mass2`; it's meaningless. mass(k) must be written, and then in a subsequent loop (looks like that loop was erased, but I think it needs to be restored), the mass diff should be computed.